File: src/main/java/eu/happybe/openapi/form/types/SimpleForm.java
package eu.happybe.openapi.form.types;

import eu.happybe.openapi.form.Form;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class SimpleForm extends Form {

    public SimpleForm(String title, String content) {
        this.getData().put("type", "form");
        this.getData().put("title", title);
        this.getData().put("content", content);
        this.getData().put("buttons", new ArrayList<Map<String, Object>>());
    }

    public SimpleForm(String title) {
        this(title, "Form Content");
    }

    public SimpleForm() {
        this("Form Title");
    }

    public SimpleForm addButton(String text) {
        Map<String, Object> buttonData = new HashMap<>();
        buttonData.put("text", text);

        ((List<Map<String, Object>>)this.getData().get("buttons")).add(buttonData);
        return this;
    }
}

File: src/main/java/eu/happybe/openapi/form/types/SimpleForm.kt
package eu.happybe.openapi.form.types

import eu.happybe.openapi.form.Form
import java.util.*

class SimpleForm @JvmOverloads constructor(title: String? = "Form Title", content: String? = "Form Content") : Form() {
    fun addButton(text: String): SimpleForm {
        val buttonData: MutableMap<String, Any> = HashMap()
        buttonData["text"] = text
        (data["buttons"] as MutableList<Map<String, Any>?>).add(buttonData)
        return this
    }

    init {
        data["type"] = "form"
        data["title"] = title
        data["content"] = content
        data["buttons"] = ArrayList<Map<String, Any>>()
    }
}